docs/proposals/deployment.md

```diff
@@ -0,0 +1,119 @@
+# Deployment
+
+## Abstract
+A proposal for implementing a new resource - Deployment - which will enable
+declarative config updates for Pods and ReplicationControllers.
+
+Users will be able to create a Deployment, which will spin of
+a ReplicationController to bring up the desired pods.
+Users can also target the Deployment at existing ReplicationControllers, in
+which case the new RC will replace the existing ones.
```

Comment:
How would the replace happen?  Might want to detail that a bit more about the replacement below or put this in future